### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by carmel4a** [MaSzyna-Reloaded/MaSzyna-API-wrapper#40] [General]
We need to pick license for our project, considering original MaSzyna licenses

**Comment by carmel4a** [comment]
- Useful links:
  - MaSzyna project license: https://eu07.pl/theme/Maszyna/dokumentacja/inne/readme_pliki/licencja.html
  - MaSzyna executable source code license: htts://github.com/eu07/maszyna/blob/master/LICENSE
  - one of discussions about license on eu07 forum: https://eu07.pl/forum/index.php/topic,26613.0.html
- Current proposals: GNU Public License, MPLv2

**Comment by marcinn** [comment]
GPT says:

The MaSzyna Train Simulator's source code is licensed under the Mozilla Public License 2.0 (MPL-2.0). This license permits modification and redistribution of the code under certain conditions. Notably, MPL-2.0 is a file-level copyleft license, meaning that any modified files must remain under MPL-2.0, but you can combine them with other files under different licenses in a larger project.

Key Points Regarding MPL-2.0:

    Modification and Redistribution: You are allowed to modify the source code and distribute your modifications, provided you comply with the MPL-2.0 terms.

    Licensing of Modified Files: Any modified MPL-2.0–licensed files must retain the MPL-2.0 license. You cannot relicense these files under (L)GPL or any other license.

    Larger Works: You can include MPL-2.0–licensed files in a larger project that uses different licenses for other parts. However, the MPL-2.0–licensed files must remain under MPL-2.0.

Implications for Your Project:

    Creating a Derivative Project: You can modify parts of the MaSzyna source code and include them in your project. The modified files must remain under MPL-2.0.

    Licensing Your Project: While the modified MPL-2.0 files must retain their original license, you can license the rest of your project under (L)GPL. This approach results in a mixed-license project, where MPL-2.0–licensed files coexist with (L)GPL–licensed files.

Considerations:

    Compatibility: Ensure that combining MPL-2.0–licensed code with (L)GPL–licensed code does not violate the terms of either license. While MPL-2.0 is generally compatible with GPL, careful attention is required to maintain compliance.

    Distribution: When distributing your project, you must provide the source code for the MPL-2.0–licensed files and comply with both MPL-2.0 and (L)GPL requirements for their respective parts.

Conclusion:

You can modify parts of the MaSzyna source code and include them in a derivative project. The modified files must remain under the MPL-2.0 license, but you can license other parts of your project under (L)GPL, resulting in a mixed-license project. Ensure compliance with the terms of both licenses when distributing your project.
